ui/src/components/SearchInput/SearchInput.vue

```diff
@@ -1,107 +1,110 @@
-          styling="light"
-          slot="after"
-          icon="search"
-          @click="handleSearchClick"
-        />
-      </InputGroup>
-    </div>
-  </div>
+  <FdComboboxBase class="fd-search-input">
+    <template #input="{showCompletions, hideCompletions}">
+      <FdInput
+        :value="currentPredicate"
+        :placeholder="placeholder"
+        :compact="compact"
+        @focus.native="showCompletions"
+        @update="setCurrentPredicate"
+        @change="$emit('change', $event)"
+        @keyup.native.esc="hideCompletions"
+      />
+    </template>
+
+    <template #after="{ toggleCompletions }">
+      <FdButton
+        @click="toggleCompletions"
+        :compact="compact"
+        icon="search"
+        styling="light"
+      />
+    </template>
+
+    <template #default="{ hideCompletions }">
+      <FdCompletionList
+        v-if="hasMatches"
+        :completions="completionsMatchingPredicate"
+        :predicate="currentPredicate"
+        @select="
+          completion => setCurrentPredicate(completion) || hideCompletions()
+        "
+      />
+      <template v-else>
+        <slot name="empty">
+          <p class="fd-has-text-align-center" v-type:-1 v-padding:tiny>
+            nothing here
+          </p>
+        </slot>
+      </template>
+    </template>
+  </FdComboboxBase>
 </template>
 
-<script lang="ts">
-import { Uid, mixins } from "@/mixins";
-import { Popover } from "@/components/Popover";
-import { Button } from "@/components/Button";
-import { Input, InputGroup } from "@/components/Form";
+<script>
+import { Uid } from "@/mixins";
+import FdButton from "./../Button/Button.vue";
+import FdInput from "./../Form/Controls/Input.vue";
+import FdCompletionList from "./CompletionList.vue";
+import FdComboboxBase from "./../ComboboxBase/ComboboxBase.vue";
 
-export default mixins(Uid).extend({
+export default {
   name: "FdSearchInput",
-  components: { Popover, Button, InputGroup, Input },
+  mixins: [Uid],
+  model: {
+    prop: "predicate",
+    event: "update"
+  },
+  components: {
+    FdComboboxBase,
+    FdCompletionList,
+    FdButton,
+    FdInput
+  },
   props: {
-    value: { type: String, default: "" },
+    predicate: { type: String, default: null },
+    completions: { type: Array, default: () => [] },
     placeholder: { type: String, default: "" },
-    ariaLabel: { type: String, default: "Search Input" },
     compact: { type: Boolean, default: false }
   },
   computed: {
-    suggestionsAvailable(): boolean {
-      const list = this.$slots.default || [];
-      return list.length > 0;
-    }
-  },
-  methods: {
-    handleSearchClick() {
-      this.emitSearch();
+    hasMatches() {
+      return this.completionsMatchingPredicate.length > 0;
     },
-    setCurrentValue(newValue: string) {
-      this.searchValue = newValue;
-      this.emitSearch();
-      this.$emit("update:value", this.searchValue);
+    normalizedPredicate() {
+      console.log(this.predicate);
```

Comment:
can you remove the `console.log` ? :D

Reply:
I will remove that and add the linter rule so that this does not happen again. :D

Reply:
💯